src/cl.rs

```diff
@@ -127,6 +130,23 @@ pub(crate) enum Action {
         /// Target directory (must exist)
         dir: PathBuf,
     },
+    #[cfg(feature = "gen-man-pages")]
+    GenShellComplete {
+        #[clap(flatten)]
+        arg_group: GenShellCompleteGroup,
+    },
+}
+
+#[cfg(feature = "gen-man-pages")]
+#[derive(Debug, clap::Args)]
+#[group(required = true, multiple = true)]
+pub(crate) struct GenShellCompleteGroup {
+    /// Shell to generate for, leave empty for all
+    #[arg(short = 's', long, default_value = None)]
+    pub shell: Option<Shell>,
+    /// Target directory
+    #[arg(num_args = 1, required = false)]
+    pub dir: Option<PathBuf>,
 }
```

Comment:
Why define a new `GenShellCompleteGroup` type used only once, and then flatten? This can be simplified down to:
```suggestion
    #[cfg(feature = "gen-man-pages")]
    GenShellComplete {
        /// Shell to generate for, leave empty for all
        #[arg(short = 's', long, default_value = None)]
        shell: Option<Shell>,
        /// Target directory
        #[arg(num_args = 1, required = false)]
        dir: Option<PathBuf>,
    }
```

Reply:
What issues are you having with the simple version?

I forgot the `group` line in my suggestion above, but it seems to work when I try see https://github.com/desbma/shh/commit/a35fece76c78bd92387738a1f329c73d9c2b794c.

And it gives a coherent signature help:
```
Usage: shh gen-shell-complete <--shell <SHELL>|DIR>

Arguments:
  [DIR]  Target directory

Options:
  -s, --shell <SHELL>  Shell to generate for, leave empty for all [possible values: bash, elvish, fish, powershell, zsh]
  -h, --help           Print help
```

Which errors when no argument is given:
```
$ cargo run --all-features -- gen-shell-complete
[...]
error: the following required arguments were not provided:
  <--shell <SHELL>|DIR>

Usage: shh gen-shell-complete <--shell <SHELL>|DIR>

For more information, try '--help'
```